bittensor/utils/networking.py

```diff
@@ -92,27 +92,29 @@ def get_external_ip() -> str:
             ExternalIPNotFound (Exception):
                 Raised if all external ip attempts fail.
     """
-    # --- Try curl.
+    # --- Try ipconfig.
     try:
-        process =  os.popen('curl -s ifconfig.me', close_fds=True)
+        process =  os.popen('curl -s ifconfig.me')
         external_ip = process.readline()
         process.close()
         assert isinstance(ip_to_int(external_ip), int)
         return str(external_ip)
     except Exception:
         pass
 
-    # --- Try ipify
+    # --- Try AWS
     try:
-        external_ip = requests.get('https://api.ipify.org').text
+        external_ip = requests.get('https://checkip.amazonaws.com').text.strip()
         assert isinstance(ip_to_int(external_ip), int)
         return str(external_ip)
     except Exception:
         pass
```

Comment:
is this already ordered? Or may should have AWS as the first option?

Reply:
Not yet, it should be AWS -> ifconfig -> ipinfo. AWS should have the most consistency.